src/components/common/CommentList/index.js

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { List, Skeleton, Modal, Button, Row } from 'antd';
+import { FormattedRelative, FormattedMessage } from 'react-intl';
+
+import { prepareData } from '../../../api/util/helpers';
+import CommentCreateForm from './CommentCreateForm';
+
+// TODO think about CSSinJS for styles
+const formButton = {
+  type: 'primary',
+  ghost: true,
+  style: {
+    border: 'none',
+    padding: 0,
+    height: 'auto',
+    boxShadow: 'none'
+  }
+};
+
+class CommentList extends React.Component {
+  state = {
+    list: this.props.data || [],
+    visible: false
+  };
+
+  showModal = () => {
+    this.setState({ visible: true });
+  };
+
+  saveFormRef = (formRef) => {
+    this.formRef = formRef;
+  };
+
+  handleCancel = () => {
+    this.setState({ visible: false });
+  };
+
+  deleteComment = item => {
+    // I have never liked assigning THIS to SELF (((
```

Comment:
You don't need to. Use an arrow function in line 46 instead. Unless I'm missing something.

Reply:
thinking about it - it would probably be more readable moving the deletion to a separate function anyhow?